staged-documentation.md

```diff
@@ -0,0 +1,282 @@
+## Overriding bean factories
+
+When a bean does not contain all the data required for the template, you can override its factory to produce a subclass of the bean.
+
+In this example shows how to add a field `lastUpdated` to the `ProductDetailPageContent`. This procedure is not limited to page class beans.
+
+First override the bean to add the field including getters and setters:
+
+```java
+public class MyProductDetailPageContent extends ProductDetailPageContent {
+    private String lastUpdated;
+
+    public String getLastUpdated() {
+        return lastUpdated;
+    }
+
+    public void setLastUpdated(final String lastUpdated) {
+        this.lastUpdated = lastUpdated;
+    }
+}
+```
+
+Then override the factory for the original bean:
+
+```java
+public class MyProductDetailPageContentFactory extends ProductDetailPageContentFactory {
+    //here you can use dependency injection to get instances
+    @Inject
+    private UserContext userContext;
+
+    //do not call here super.create because you would get a bean of the original class and not the subclass
+    @Override
+    public ProductDetailPageContent create(final ProductProjection product, final ProductVariant variant) {
+        final MyProductDetailPageContent bean = new MyProductDetailPageContent();
+        initialize(bean, product, variant);
+        return bean;
+    }
+
+    //the new initialize method contains as first parameter the bean of your subclass and then the parameters of the create method
+    protected final void initialize(final MyProductDetailPageContent bean, final ProductProjection product, final ProductVariant variant) {
```

Comment:
I know this is the proper way of doing it if you want to enable subclasses. But it's easier to explain/understand if you just put the missing information in the `create` method. If they plan on using subclasses they will probably realize how to fix it on their own or they will check our class as inspiration (since they know it is possible to do because we are doing it). I would try to keep the explanations as simple as possible (as in the trainings), it gives the feeling it is very simple and leaves a positive impression. We can always comment in text later that a good practice to allow subclasses would be to follow the same approach as we do, but that it is unnecessary for simple cases.